### PR TITLE
Added ability to use scientific notation

### PR DESCRIPTION
**Pull request by MHumm**
As proposed in Mantis issue 6545:
[http://issuetracker.delphi-jedi.org/view.php?id=6545#c21743](url)

### CODE REVIEW

File: jvcl/run/JvInterpreterConst.pas [hunk 1]
```diff
@@ -43,6 +43,9 @@ interface
   StIdFirstSymbols = ['_', 'A'..'Z', 'a'..'z'];
   StConstSymbols = ['0'..'9', 'A'..'F', 'a'..'f'];
   StConstSymbols10 = ['0'..'9'];
+  StConstE=['E','e'];   //lht+
```
Comment (obones) [inline jvcl/run/JvInterpreterConst.pas]: The comment with the contributor name at the end should be removed

File: jvcl/run/JvInterpreterParser.pas [hunk 2]
```diff
@@ -433,6 +433,7 @@ function TokenTyp(const Token: string): TTokenKind;
   T1: Char;
   Ci: Char;
   Point: Boolean;
+  sci_nota:Boolean;
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: The name of this variable does not respect the coding guide. Please rename it to `IsScientificNotation`

File: jvcl/run/JvInterpreterParser.pas [hunk 3]
```diff
@@ -499,19 +500,23 @@ function TokenTyp(const Token: string): TTokenKind;
         Any: { !!LABEL!! }
 
         Point := False;
+        sci_nota := False;
         for I := 1 to L1 do
         begin
           Ci := Token[I];
+          if CharInSet(Ci,StConstE) then
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: Missing space after the comma

File: jvcl/run/JvInterpreterParser.pas [hunk 4]
```diff
@@ -587,7 +592,7 @@ function TJvInterpreterParser.Token: string;
   F1: PChar;
   I: Integer;
   PrevPoint:boolean;
-//  PointCount: Integer;
+  point_occur, e_occur:Boolean;
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: Same comment about naming. Use `PointOccurred` and `ExponentOccurred`

File: jvcl/run/JvInterpreterParser.pas [hunk 5]
```diff
@@ -658,10 +666,28 @@ function TJvInterpreterParser.Token: string;
...
   { number }
   begin
-    while CharInSet(P[0], StConstSymbols10) or (P[0] = '.') do
+    while CharInSet(P[0], StConstSymbols10e) do  
     begin
-      if (P[0] = '.') and (P[1] = '.') then
-        Break;
+      if P[0] = '.' then
+      begin
+        if point_occur //lht��radix point can occur zero or one time
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: Contributor comment to be removed

File: jvcl/run/JvInterpreterParser.pas [hunk 6]
```diff
@@ -658,10 +666,28 @@ function TJvInterpreterParser.Token: string;
...
+            or not CharInSet(P[-1],StConstSymbols10) //lht��a radix point must behind a number
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: I'm pretty sure the coding style requires the "or" to be at the end of the line and that indentation is only one block (ie two spaces)

File: jvcl/run/JvInterpreterParser.pas [hunk 7]
```diff
@@ -658,10 +666,28 @@ function TJvInterpreterParser.Token: string;
...
+            or (P[1] = '.') then
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: Same as above

File: jvcl/run/JvInterpreterParser.pas [hunk 8]
```diff
@@ -658,10 +666,28 @@ function TJvInterpreterParser.Token: string;
...
+        if point_occur //lht��radix point can occur zero or one time
+            or not CharInSet(P[-1],StConstSymbols10) //lht��a radix point must behind a number
+            or (P[1] = '.') then  
+          Break;
+        point_occur:=True;
+      end
+      else if CharInSet(P[0],StConstE)  then
+      begin
+        if e_occur  //lht�� only one time ,at most
+          or  not CharInSet(P[-1],StConstSymbols10)  then //lht��E must behind a number
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: Same as above

File: jvcl/run/JvInterpreterParser.pas [hunk 9]
```diff
@@ -658,10 +666,28 @@ function TJvInterpreterParser.Token: string;
...
+      begin
+        if point_occur //lht��radix point can occur zero or one time
+            or not CharInSet(P[-1],StConstSymbols10) //lht��a radix point must behind a number
+            or (P[1] = '.') then  
+          Break;
+        point_occur:=True;
+      end
+      else if CharInSet(P[0],StConstE)  then
+      begin
+        if e_occur  //lht�� only one time ,at most
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: Same as above

File: jvcl/run/JvInterpreterParser.pas [hunk 10]
```diff
@@ -658,10 +666,28 @@ function TJvInterpreterParser.Token: string;
...
+          or  not CharInSet(P[-1],StConstSymbols10)  then //lht��E must behind a number
+          Break;
+        e_occur := True;
+      end
+      else if CharInSet(P[0],StConstPlusSub) then
+      begin
+        if not CharInSet(P[-1],StConstE) then //lht: +/- must behind E
```
Comment (obones) [inline jvcl/run/JvInterpreterParser.pas]: Please remove contributor comment

**Comment by MHumm** [comment]
Superseded by #115

**Pull request closed without merging**